<?xml version="1.0"?>
<launch>=

    <!-- ============================ SETTINGS ============================= -->

    <!-- Node frequency: the frequency [Hz] which rate.sleep() function will
    attempt to maintain in the main_loop. -->
    <arg 
    name="node_frequency" 
    default="100"/>

    <!-- Controller side: can be either "right" or "left". -->
    <arg 
    name="controller_side" 
    default="right"/>

    <!-- Max speed fraction: maximum chest motion speed (0.01 - 1.0), which
    represents the fraction of the maximum chest speed. -->
    <arg 
    name="max_speed_fraction" 
    default="1.0"/>

    <!-- Enable chest compensation for Kinova: Kinova EE stabilization based on
    the chest position. -->
    <arg 
    name="enable_chest_compensation_for_kinova" 
    default="false"/> 


    <!-- If any of dependency or optional subfiles should not be launched (or
    are launched separately) set corresponding argument to "false". -->
    <arg 
    name="launch_chest" 
    default="true"/>
    <arg 
    name="launch_controller_feedback" 
    default="true"/>

    <!-- ======================== END OF SETTINGS ========================== -->

    <!-- Chest modules (dependency): -->
    <group if="$(arg launch_chest)">
        <include 
        file="$(find gopher_ros_clearcore)/launch/chest.launch"/>
    </group>

    <!-- Oculus controller feedback (dependency): -->
    <group if="$(arg launch_controller_feedback)">
        <include 
        file="$(find oculus_ros)/launch/controller_feedback.launch">
            <arg 
            name="controller_side" 
            value="$(arg controller_side)"/>
        </include>
    </group>

    <!-- Oculus mapping: -->
    <node 
    pkg="gopher_ros_clearcore" 
    type="oculus_chest_mapping.py" 
    name="oculus_chest_mapping" 
    output="screen">
        <param 
        name="node_frequency" 
        value="$(arg node_frequency)"/>
        <param 
        name="controller_side" 
        value="$(arg controller_side)"/>
        <param 
        name="max_speed_fraction" 
        value="$(arg max_speed_fraction)"/>
        <param 
        name="enable_chest_compensation_for_kinova" 
        value="$(arg enable_chest_compensation_for_kinova)"/>
    </node>
</launch>